src/net_processing.cpp

```diff
@@ -1017,7 +1024,15 @@ void PeerManagerImpl::ReattemptInitialBroadcast(CScheduler& scheduler) const
 
         if (tx != nullptr) {
             LOCK(cs_main);
-            RelayTransaction(txid, tx->GetWitnessHash(), m_connman);
+            // We re-relay these transactions via reconciliation (same as for initial broadcast),
+            // instead of flooding, to hide the origin of transactions.
+            // Since in Erlay transactions are flooded only inbound->outbound,
+            // non-reachable nodes never flood transactions from other nodes (they have no inbounds).
+            // Thus, making them flood these would tell a receiver that these indeed belong to the
+            // flooding non-reachable nodes. Instead, we relay them via reconciliation,
+            // in which case a receiver can't distinguish them from transactions we reconciled
+            // with some other peer.
+            RelayTransaction(txid, tx->GetWitnessHash(), m_connman, false);
```

Comment:
>Non-reachable nodes may open connections to outbound not supporting reconciliations, thus relying on flooding for transaction announcements, at least until Erlay is well-deployed.

Sure. But for reconciliation conns, the current policy is to flood only what was received via inbound. So if we rebroadcast to a reconciling peer via flooding, they will know it's rebroadcast, because non-reachable never receive any stuff via inbound (by definition)